serverless-plugin/index.js

```diff
@@ -145,15 +145,19 @@ class ServerlessPlugin {
 
     assertServerlessPackageVersionsMatch(directory, packageName) {
         // Check that the Haskell package version corresponds to our own
-        const haskellPackageVersion = this.runStackOutput(
+        const stackDependencies = this.runStackOutput(
             directory,
             [
-                'exec', '--',
-                'ghc-pkg',
-                'field', PACKAGE_NAME, 'version',
-                '--simple-output'
+                'ls',
+                'dependencies',
             ]
-        );
+        ).split("\n");
+        const haskellPackageVersions = stackDependencies.filter(dep => dep.startsWith(`${PACKAGE_NAME} `));
+        if (haskellPackageVersions.length === 0) {
+            this.serverless.cli.log(`Could not find ${PACKAGE_NAME} in stack's dependencies. Make sure ${PACKAGE_NAME} is listed as a dependency in your .cabal file, or as an extra-dep in your stack.yaml.`);
```

Comment:
Cabal files don't matter here (stack is required, and the build will fail if serverless-haskell is in the Cabal dependencies but not in stack). So to get serverless-haskell, you need to either use LTS 12+, or add it to extra-deps.